win32more/xaml.py

```diff
@@ -0,0 +1,174 @@
+from ctypes import (
+    POINTER,
+    WINFUNCTYPE,
+    Structure,
+    WinError,
+    addressof,
+    c_void_p,
+    cast,
+    py_object,
+    sizeof
+)
+
+from win32more import FAILED, Guid, UInt32
+from win32more._winrt import WinRT_String, SZArray
+from win32more.mddbootstrap import MddBootstrapInitialize2, MddBootstrapShutdown, MddBootstrapInitializeOptions_OnNoMatch_ShowUI
+from win32more.Windows.Win32.Foundation import HRESULT, S_OK
+from win32more.Windows.Win32.Storage.Packaging.Appx import PACKAGE_VERSION
+from win32more.Windows.Win32.System.WinRT import HSTRING, IInspectable, TrustLevel
+from win32more.Windows.UI.Xaml.Interop import TypeName
+from win32more.Microsoft.UI.Xaml import Application, IApplicationOverrides, LaunchActivatedEventArgs
+from win32more.Microsoft.UI.Xaml.Controls import XamlControlsResources
+from win32more.Microsoft.UI.Xaml.Markup import IXamlType, XmlnsDefinition, IXamlMetadataProvider
+from win32more.Microsoft.UI.Xaml.XamlTypeInfo import XamlControlsXamlMetaDataProvider
+
+class XamlApplicationImpl(Structure):
+    _fields_ = [("lpvtbl", POINTER(c_void_p)), ("lpvtbl2", POINTER(c_void_p)), ("comptr", py_object)]
+
+    def __init__(self, comptr):
+        self.lpvtbl = (c_void_p * 7)()
+        self.lpvtbl[0] = cast(self.QueryInterface, c_void_p)
+        self.lpvtbl[1] = cast(self.AddRef, c_void_p)
+        self.lpvtbl[2] = cast(self.Release, c_void_p)
+        self.lpvtbl[3] = cast(self.GetIids, c_void_p)
+        self.lpvtbl[4] = cast(self.GetRuntimeClassName, c_void_p)
+        self.lpvtbl[5] = cast(self.GetTrustLevel, c_void_p)
+        self.lpvtbl[6] = cast(self.OnLaunched, c_void_p)
+        self.lpvtbl2 = (c_void_p * 9)()
+        self.lpvtbl2[0] = cast(self.QueryInterface, c_void_p)
+        self.lpvtbl2[1] = cast(self.AddRef, c_void_p)
+        self.lpvtbl2[2] = cast(self.Release, c_void_p)
+        self.lpvtbl2[3] = cast(self.GetIids, c_void_p)
+        self.lpvtbl2[4] = cast(self.GetRuntimeClassName, c_void_p)
+        self.lpvtbl2[5] = cast(self.GetTrustLevel, c_void_p)
+        self.lpvtbl2[6] = cast(self.GetXamlType, c_void_p)
+        self.lpvtbl2[7] = cast(self.GetXamlTypeByFullName, c_void_p)
+        self.lpvtbl2[8] = cast(self.GetXmlnsDefinitions, c_void_p)
+        self.comptr = py_object(comptr)
+
+    @WINFUNCTYPE(HRESULT, c_void_p, POINTER(Guid), POINTER(c_void_p))
+    def QueryInterface(this, riid, ppvObject):
+        self = cast(this, POINTER(XamlApplicationImpl)).contents
+        return self.comptr.QueryInterface(riid, ppvObject)
+
+    @WINFUNCTYPE(UInt32, c_void_p)
+    def AddRef(this):
+        self = cast(this, POINTER(XamlApplicationImpl)).contents
+        return self.comptr.AddRef()
+
+    @WINFUNCTYPE(UInt32, c_void_p)
+    def Release(this):
+        self = cast(this, POINTER(XamlApplicationImpl)).contents
```

Comment:
`this` is `pointer(lpvtbl2)` when invoked as IXamlMetadataProvider.
In this case `cast(this POINTER(XamlApplicationImpl)` is invalid.

Reply:
Thank you. I fixed it with a little bit dirty way.